src/Converters/Models/MatrixBlockType.php

```diff
@@ -38,13 +38,36 @@ public function getRecordDefinition(Model $record): array
      */
     public function saveRecord(Model $record, array $definition): bool
     {
-        $context = 'matrixBlockType:'.$record->id;
-        $existingFields = Craft::$app->fields->getAllFields($context);
+        $this->resetCraftMatrixServiceBlockTypesCache();
+
+        // Get existing fields by block type handle
+        $existingFields = [];
+        $existingBlockTypes = Craft::$app->matrix->getBlockTypesByFieldId($record->fieldId);
+        foreach ($existingBlockTypes as $existingBlockType) {
+            if ($existingBlockType->handle == $definition['attributes']['handle']) {
+                $existingFields = $existingBlockType->getFields();
```

Comment:
$record should already be the existingBlockType, can't you just call getFields on it?

Reply:
Yeah I can, but for some reason it doesn't skip fields the second time when I use that - while this does. I will dive deeper into it and find the real reason.